### ISSUE TITLE
Keyring issue with KDE on Wayland

### ISSUE THREAD

**Issue by MggMuggins** [pithos/pithos#554]
I'm starting the move to KDE on Wayland, but Pithos isn't running because of a keyring issue. I have the keyring unlocked (SDDM unlocks libsecret "login" on login), and it works just fine in X. Here's the Error Dialog Text:

**Error calling StartServiceByName for org.freedesktop.secrets: Timeout was reached**
    You need to install a service such as gnome-keyring

And here's the log running from the command line:
```
ERROR - util:on_get_finish:88 - Failed to get Secret.Service, Error: g-io-error-quark: Error calling StartServiceByName for org.freedesktop.secrets: Timeout was reached (24)
ERROR - pithos:on_keyring_unlocked:280 - You need to install a service such as gnome-keyring. Error: g-io-error-quark: Error calling StartServiceByName for org.freedesktop.secrets: Timeout was reached (24)
```
I'd really appreciate a fix for this, since I'd like to use Plasma on Wayland as much as I can.

**Comment by MggMuggins** [comment]
Beginning to realize that this may not be a Pithos issue. Sorry about that, if I find anything that points to something else, I'll close this.

**Comment by JasonLG1979** [comment]
Looks like an upstream issue or possibly a configuration problem on your system? 

You've had [keyring issues in the past](https://github.com/pithos/pithos/issues/511). How are you starting GNOME Keyring and can you verify that it starts in the wayland session and is running when you try to start Pithos.

**Comment by TingPing** [comment]
It *should* be dbus activated and not need to manually start.

**Comment by MggMuggins** [comment]
I would agree with TingPing, and sddm unlocks the Keyring. Like I said, it works with X. What process am I looking for?

**Comment by TingPing** [comment]
libsecret talks over the user bus to `org.freedesktop.secrets` which should be the provided by `gnome-keyring` and launch the `gnome-keyring-daemon` process.

**Comment by JasonLG1979** [comment]
> It should be dbus activated and not need to manually start.

@TingPing see https://wiki.archlinux.org/index.php/GNOME/Keyring#Using_the_keyring_outside_GNOME

I had a similar issue with the KDE Fedora Spin.

**Comment by TingPing** [comment]
@JasonLG1979 I sort of just want to drop libsecret, it is a pita and has its own security problems.

**Comment by JasonLG1979** [comment]
And replace it with what?

**Comment by TingPing** [comment]
Just storing it manually in a private file. With Flatpak it actually *improves* security because `libsecret` isn't designed for sandboxing and exposes all stored passwords to all applications. Outside of Flatpak the situation is largely the same since other unprivileged users can't read it and FDE is required for any further security anyway.

**Comment by JasonLG1979** [comment]
> Just storing it manually in a private file. With Flatpak it actually improves security because libsecret isn't designed for sandboxing and exposes all stored passwords to all applications. Outside of Flatpak the situation is largely the same since other unprivileged users can't read it and FDE is required for any further security anyway.

So basically go back to the way it was before except with better file permissions?

**Comment by TingPing** [comment]
> So basically go back to the way it was before except with better file permissions?

Yes.

**Comment by JasonLG1979** [comment]
Where would you want the creds stored? .config?

**Comment by JasonLG1979** [comment]
If we're going to store the creds in the user's home it would be nice if the creds were not human readable for the sake of dissuading your average lookie-loo.

**Comment by TingPing** [comment]
Eh, seems a bit false security through obscurity kinda thing.

**Comment by JasonLG1979** [comment]
> Eh, seems a bit false security through obscurity kinda thing.

Like I said it's just to keep someone who has access to the file from casually opening it and reading the creds. It's not going to stop someone who really wants the creds. If they have access to read the file there's nothing that stops them from figuring out the creds no matter how it's obfuscated. That's really true of any way we would protect the creds, short of forcing the user to enter a key to unlock the creds every time we start which would be pointless as at that point we may as well not even store the creds at all.

**Comment by JasonLG1979** [comment]
At this point how GNOME keyring works is false security.

user logs in > keyring is unlocked and all creds are available as plaintext to whoever wants to read them.

**Comment by JasonLG1979** [comment]
Scenario where a little bit of obscurity is a good thing:

You forget you're still logged into your computer. You're asshat roommate jumps on and snoops around your computer.

plaintext:
asshat has your Pandora creds.

obfuscated:
He's also a dumbass so he has no idea what your Pandora creds are because they're base64 encoded.

**Comment by TingPing** [comment]
If we are storing the authtoken not the password that is already obscure enough that random joe can't use it.

**Comment by jayschieber** [comment]
I had the same keyring issue when running the flatpak version of Pithos (1.4.1) on Kubuntu 18.04.
But it was fixed by just installing seahorse.
